backup/backup.go

```diff
@@ -20,6 +21,132 @@ import (
 	"go.uber.org/zap"
 )
 
+type BucketManifestWriter struct {
+	ts *tenant.Service
+	mc *meta.Client
+}
+
+func NewBucketManifestWriter(ts *tenant.Service, mc *meta.Client) BucketManifestWriter {
+	return BucketManifestWriter{
+		ts: ts,
+		mc: mc,
+	}
+}
+
+// WriteManifest writes a bucket manifest describing all of the buckets that exist in the database.
+// It is intended to be used to write to an HTTP response after appropriate measures have been taken
+// to ensure that the request is authorized.
+func (b BucketManifestWriter) WriteManifest(ctx context.Context, w io.Writer) error {
+	bkts, _, err := b.ts.FindBuckets(ctx, influxdb.BucketFilter{})
+	if err != nil {
+		return err
+	}
+
+	l := []influxdb.BucketMetadataManifest{}
```

Comment:
Could do this as a small optimization:
```
l := make([]influxdb.BucketMetadataManifest, 0, len(bkts))
```

Reply:
Nice! I did that for this and the other similar functions.